src/core/qgsogrutils.cpp

```diff
@@ -221,6 +221,17 @@ QVariant QgsOgrUtils::getOgrFeatureAttribute( OGRFeatureH ogrFet, const QgsField
           value = QDateTime( QDate( year, month, day ), QTime( hour, minute, second ) );
       }
       break;
+
+      case QVariant::ByteArray:
+      {
+        int size = 0;
+        const GByte *b = OGR_F_GetFieldAsBinary( ogrFet, attIndex, &size );
+        QByteArray ba = QByteArray::fromRawData( reinterpret_cast<const char *>( b ), size );
+        ba.detach();
```

Comment:
Why the detach?

Reply:
http://doc.qt.io/qt-5/qbytearray.html#fromRawData

Reply:
I'll add a comment here. It's an odd API decision on the qt side.